framework/packages/abstract-client/src/builder.rs

```diff
@@ -128,26 +128,28 @@ impl<Chain: CwEnv> AbstractClientBuilder<Chain> {
     }
 
     /// Deploy abstract with current configuration
-    pub fn build(&self) -> AbstractClientResult<AbstractClient<Chain>> {
-        let abstr =
-            Abstract::deploy_on(self.chain.clone(), self.chain.sender_addr().into_string())?;
-        self.update_ans(&abstr)?;
+    pub fn build(
+        &self,
+        abstract_admin: Chain::Sender,
+    ) -> AbstractClientResult<AbstractClient<Chain>> {
+        let abstr = Abstract::deploy_on(self.chain.clone(), abstract_admin.clone())?;
+        self.update_ans(&abstr, abstract_admin)?;
```

Comment:
A bit painful, can we avoid it?

Reply:
what's painful exactly there ?

Reply:
> what's painful exactly there ?

Passing sender on client builder. I guess it's fine as long as it's easy to get this sender

Reply:
We should be able to add a `build_test` fn that excludes sender and applies for `Chain = Mock<A, S>`?